DB/src/main/java/io/deephaven/db/v2/locations/parquet/topage/ToDBDateTimePage.java

```diff
@@ -3,39 +3,77 @@
 import io.deephaven.db.tables.dbarrays.DbArray;
 import io.deephaven.db.tables.dbarrays.DbArrayDirect;
 import io.deephaven.db.tables.utils.DBDateTime;
+import io.deephaven.db.tables.utils.DBTimeUtils;
 import io.deephaven.db.v2.sources.chunk.Attributes;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
 import org.jetbrains.annotations.NotNull;
 
-public class ToDBDateTimePage<ATTR extends Attributes.Any> extends ToLongPage<ATTR> {
+import java.util.function.LongUnaryOperator;
 
-    private static ToDBDateTimePage INSTANCE = new ToDBDateTimePage<>();
+public abstract class ToDBDateTimePage<ATTR extends Attributes.Any> extends ToLongPage<ATTR> {
 
-    public static <ATTR extends Attributes.Any> ToDBDateTimePage<ATTR> create(@NotNull Class<?> nativeType) {
+    private static <ATTR extends Attributes.Any> ToDBDateTimePage<ATTR> fromUnitToNanos(final LongUnaryOperator convertUnitToNanos) {
+        return new ToDBDateTimePage<ATTR>() {
+            private long toNanos(final long v) {
+                return convertUnitToNanos.applyAsLong(v);
```

Comment:
I know you were doing this to appeal to my sensibilities, but I suspect the abstract method is slightly better/more natural.

Reply:
I don't love the anonymous inner class approach. I think the best design is to add an abstract `toDateTime` method as you did before, and use it in `toDbArray` for all `ToDBDateTimePage` impls, and use it in an overriden `convertResult` in the millis and micros implementations. What you had before this round was better, just needed to add the `convertResult` overrides.